src/test/ui/typeck/slow-lhs-suggestion.rs

```diff
@@ -0,0 +1,26 @@
+fn main() {
+    1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1;
```

Comment:
This is quite a bit longer than the reported test case. I assume the exponential blowup of type checking this prior to the fix would prevent this from regressing in the future?

Reply:
yeah, i just wanted to be certain it was being caught.

Reply:
Cool, that's what I figured!